doc/doxygen/src/kconfig/kconfig.md

```diff
@@ -0,0 +1,366 @@
+sure that when cleaning an application you call `make clean && make all`,
+instead of both targets on a single command.
+
+---
+# Integration into the build system    {#kconfig-integration-into-build-system}
+
+The integration of Kconfig into the build system is mainly done in
+`makefiles/kconfig.mk`.
+
+## Steps during the build process                {#kconfig-steps-build-process}
+
+![Ouput of every step](kconfig_integration.svg)
+
+### 0. Module dependency resolution
+Currently, the resolution of module dependencies is being performed by the build
+system where all the used modules and packages end up listed in the `USEMODULE`
+make variables. In the next phases of integration we plan to resolve dependencies
+using Kconfig.
+
+
+#### Input
+- Makefiles.
+
+#### Output
+- `USEMODULE` and `USEPKG` variables.
+
+### 1. Module listing
+The list of modules needed for the particular build is dumped into the
+`$ (GENERATED_DIR)/Kconfig.dep` file, where each module is translated into a
+Kconfig symbol as documented in [Appendix A](#kconfig-appendix-a).
+
+#### Input
+- `USEMODULE` and `USEPKG` variables
+
+#### Output
+- `$ (GENERATED_DIR)/Kconfig.dep` file
+
+### 2. Merging all configuration sources
+In this step configuration values are taken from multiple sources and merged
+into a single `merged.config` configuration file. This file is temporary and is
+removed on clean. If the user needs to save a particular configuration
+set, a backup has to be saved (this can be done using the menuconfig interface)
+so it can be loaded later in this step.
+
+To accomplish merging of multiple input files, the `mergeconfig` script is used.
+Note that **the order matters**: existing configuration values are merged in the
+order expressed in the input section, where the last value assigned to a
+parameter has the priority. Any configuration file that is not present will be
+ignored. If no configuration files are available all default values will be
+applied.
+
+`merged.config` is the only configuration input for the `autoconf.h` generation in
+step 4.
+
+#### Input
+- Optional:
+    - `$ (APPDIR)/app.config`: Application specific default configurations.
+    - `$ (APPDIR)/user.config`: Configurations saved by user.
+
+#### Output
+- `$ (GENERATED_DIR)/merged.config` file.
+
+### 3. Menuconfig execution (optional)
+Menuconfig is a graphical interface for software configuration. It is used for
+the configuration of the Linux kernel. This section explains the process
+that occurs when RIOT is being configured using the menuconfig interface.
+
+The main `Kconfig` file is used in this step to show the configurable
+parameters of the system. Kconfig will filter not available parameters (i.e.
+parameters exposed by modules that are not being used) based on the file
+`$ (GENERATED_DIR)/Kconfig.dep` generated in step 1.
+
+By default, during the transition phase module configuration via Kconfig will
+not be enabled, and the user should explicitly enable this (see
```

Comment:
This sentence reads a bit bumpy. Maybe something like: "During the transition phase, the user needs to enable Kconfig explicitly by setting the tick in the checkbox". Then, leave a note what is the opposite case: If not enabled, the fallback configuration from header files is applied.